### ISSUE TITLE
defining a class that extends an abstract class but does not define all of its methods should be an error

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#109]
like `reportAbstractUsage` but on the definition instead of the usage.

explicitly extending both the abstract class and the `ABC` class should suppress the error, for cases where you intend for the class to still be abstract

**Comment by KotlinIsland** [comment]
Handle implementations getting shadowed by the MRO
```py
from typing import Sequence

from basedtyping import T


class MyList(Sequence[T]):
    pass


class MyListImpl(MyList[T], list[T]):
    pass


print(len(MyListImpl((1, 2, 3))))
```

**Comment by DetachHead** [comment]
the fact that there are abstract methods with implementations that don't raise exceptions when called and instead return a completely invalid value is absurd
```py
class Sized(metaclass=ABCMeta):

    __slots__ = ()

    @abstractmethod
    def __len__(self):
        return 0
```
